### PR TITLE
Assignment #3: Test Coverage

### PR DESCRIPTION
**Pull request by towns**
I think this is mostly right. Looking forward to any feedback. Thanks for doing this class. Its been fun so far!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+test:
+	@node node_modules/lab/bin/lab -a code
+test-cov:
+	@node node_modules/lab/bin/lab -a code -t 100 -L
+test-cov-html:
+	@node node_modules/lab/bin/lab -a code -r html -o coverage.html
+changelog:
```
Comment (AdriVanHoudt) [inline Makefile]: not part of the assignment :p

Comment (towns) [inline Makefile]: hmm, I must be confused. I thought I was satisfying this requirement:

Add the standard hapi.js Makefile to make it easy to generate other types of reports (e.g. HTML coverage report).

Comment (AdriVanHoudt) [inline Makefile]: he meant like your test-cov-html, you can also generate the report in other forms, he didn't say anything about changelog generation

File: lib/index.js [hunk 2]
```diff
@@ -10,19 +10,20 @@ var Version = require('./version');
...
+        callback = portToUse;
+        portToUse = undefined;
+    }
 
     var server = new Hapi.Server();
-    server.connection({ port: 8000 });
+    server.connection({ port: portToUse || 8000 });
     server.register(Version, function (err) {
-
         Hoek.assert(!err, err);
```
Comment (TheAlphaNerd) [inline lib/index.js]: no need to assert anymore, this will cause your app to throw if there is an error

Comment (TheAlphaNerd) [inline lib/index.js]: this error should be handled via the callback

File: lib/index.js [hunk 3]
```diff
@@ -10,19 +10,20 @@ var Version = require('./version');
 var internals = {};
 
 
-internals.init = function () {
+exports.init = function (portToUse, callback) {
+    if (typeof portToUse === 'function') {
```
Comment (TheAlphaNerd) [inline lib/index.js]: space at beginning of function

File: lib/start.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Hoek = require('hoek');
+
+Server = require('./index');
+
+Server.init(8000, function(err, server) {
+    Hoek.assert(!err, err);
```
Comment (TheAlphaNerd) [inline lib/start.js]: missing space at begining of function

File: package.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "hueniversity",
-  "version": "0.0.2",
+  "version": "0.0.3",
   "description": "Community learning experiment",
-  "main": "lib/index.js",
+  "main": "lib/start.js",
```
Comment (TheAlphaNerd) [inline package.json]: start should be living in the base of the project not in lib.  It is a consumer of lib

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+var lab = exports.lab = Lab.script();
+
+var Server = require('../lib');
+
+
+lab.test('init with specific port', function (done) {
+
+    Server.init(5000, function(err, server) {
+
+        Hoek.assert(!err, err);
```
Comment (TheAlphaNerd) [inline test/index.js]: you should be using Code to assert not Hoek

File: lib/index.js (hunk 2 above)
Comment (AdriVanHoudt) [inline lib/index.js]: unless no callback is given right?

File: lib/start.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Hoek = require('hoek');
```
Comment (simon-p-r) [inline lib/start.js]: No variable declared?

File: lib/start.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+Server = require('./index');
```
Comment (simon-p-r) [inline lib/start.js]: Same as above, no var declared

**Comment by towns** [comment]
The comment above says that I have merge conflicts that must be resolved. Any tips on what I need to do here? It looks like someone who has write access to the repo has to do it. I obviously don't.

**Comment by AdriVanHoudt** [comment]
@towns you can solve this manually merge master into your branch and resolve the conflicts locally

File: lib/index.js [hunk 9]
```diff
@@ -9,20 +9,20 @@ var Version = require('./version');
 
 var internals = {};
 
+exports.init = function(portToUse, callback) {
```
Comment (hueniverse) [inline lib/index.js]: `port` is more common name.

File: lib/index.js [hunk 10]
```diff
@@ -9,20 +9,20 @@ var Version = require('./version');
...
 
-internals.init = function () {
+    if (typeof portToUse === 'function') {
```
Comment (hueniverse) [inline lib/index.js]: This is not in the assignment.

File: lib/version.js [hunk 11]
```diff
@@ -1,23 +1,23 @@
...
 var internals = {
     response: {
-        version: require('../package.json').version
+        version: Package.version
     }
 };
 
 
-exports.register = function (server, options, next) {
+exports.register = function(server, options, next) {
```
Comment (hueniverse) [inline lib/version.js]: Purposely changing the style?

**Pull request closed without merging**